packages/protocol/ledger/conflictdag/conflictdag.go

```diff
@@ -139,53 +190,112 @@ func (b *ConflictDAG[ConflictIDType, ConflictingResourceID]) UnconfirmedConflict
 
 // SetConflictAccepted sets the ConfirmationState of the given Conflict to be Accepted - it automatically sets also the
 // conflicting conflicts to be rejected.
-func (b *ConflictDAG[ConflictID, ConflictingResourceID]) SetConflictAccepted(conflictID ConflictID) (modified bool) {
-	b.Lock()
-	defer b.Unlock()
+func (c *ConflictDAG[ConflictIDType, ResourceIDType]) SetConflictAccepted(conflictID ConflictIDType) (modified bool) {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	conflictsToReject := set.NewAdvancedSet[*Conflict[ConflictIDType, ResourceIDType]]()
 
-	rejectionWalker := walker.New[ConflictID]()
 	for confirmationWalker := set.NewAdvancedSet(conflictID).Iterator(); confirmationWalker.HasNext(); {
-		b.Storage.CachedConflict(confirmationWalker.Next()).Consume(func(conflict *Conflict[ConflictID, ConflictingResourceID]) {
-			if modified = conflict.setConfirmationState(confirmation.Accepted); !modified {
-				return
+		currentConflictID := confirmationWalker.Next()
+		conflict, exists := c.conflict(currentConflictID)
+		if !exists {
+			continue
+		}
+
+		if conflict.ConfirmationState() != confirmation.NotConflicting {
+			if !conflict.setConfirmationState(confirmation.Accepted) {
+				continue
 			}
 
-			b.Events.ConflictAccepted.Trigger(conflict.ID())
+			modified = true
 
-			confirmationWalker.PushAll(conflict.Parents().Slice()...)
+			c.Events.ConflictAccepted.Trigger(conflict)
+		}
+
+		confirmationWalker.PushAll(conflict.Parents().Slice()...)
 
-			b.Utils.forEachConflictingConflictID(conflict, func(conflictingConflictID ConflictID) bool {
-				rejectionWalker.Push(conflictingConflictID)
-				return true
-			})
+		conflict.ForEachConflictingConflict(func(conflictingConflict *Conflict[ConflictIDType, ResourceIDType]) bool {
+			conflictsToReject.Add(conflictingConflict)
+			return true
 		})
 	}
 
+	modified = c.rejectConflictsWithFutureCone(conflictsToReject) || modified
+
+	// // Delete all resolved ConflictSets (don't have a pending conflict anymore).
```

Comment:
Delete this code as we'll not be evicting conflits upon resolution.

Reply:
Can we maybe salvage it for eviction? We are not going to implement that now, but just not to rewrite this code.